plugins/node/instrumentation-kafkajs/src/propagator.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors, Aspecto
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TextMapGetter } from '@opentelemetry/api';
+
+/*
+same as open telemetry's `defaultTextMapGetter`,
+but also handle case where header is buffer,
+adding toString() to make sure string is returned
+*/
+export const bufferTextMapGetter: TextMapGetter = {
+  get(carrier, key) {
+    if (!carrier) {
+      return undefined;
+    }
+
+    const keys = Object.keys(carrier);
+
+    for (const carrierKey of keys) {
+      if (carrierKey.toLowerCase() === key) {
```

Comment:
Is `toLowerCase()` really needed here?
W3C tracecontext spec is quite clear to inject in lowercase so it seems redundant.
What about propagators which want to use mixed case keys? They wont work I guess.

Reply:
[Some propagators](https://github.com/open-telemetry/opentelemetry-java/blob/51ff6bcf335a488ff04b49514f9360f59f27677a/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/B3Propagator.java#L48-L50) inject in mixed case and [others](https://github.com/open-telemetry/opentelemetry-js/blob/d99e211a311ceefdfed67c501ea99129f5488434/packages/opentelemetry-propagator-b3/src/constants.ts) expect it in lowercase.

There's an unresolved issue for it: https://github.com/open-telemetry/opentelemetry-js/issues/3150

Reply:
ok, seems like all options are wrong :o)

What about trying both like [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/eb6e8ef6b4c72710a98ed3590392b85c8c3b79e5/propagators/opentelemetry-propagator-instana/src/InstanaPropagator.ts#L47-L51)?